src/plugins/vis_type_timeseries/public/application/components/panel_config.js

```diff
@@ -1,85 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-import PropTypes from 'prop-types';
-import React, { useState, useEffect } from 'react';
-import { TimeseriesPanelConfig as timeseries } from './panel_config/timeseries';
-import { MetricPanelConfig as metric } from './panel_config/metric';
-import { TopNPanelConfig as topN } from './panel_config/top_n';
-import { TablePanelConfig as table } from './panel_config/table';
-import { GaugePanelConfig as gauge } from './panel_config/gauge';
-import { MarkdownPanelConfig as markdown } from './panel_config/markdown';
-import { FormattedMessage } from '@kbn/i18n/react';
-
-import { FormValidationContext } from '../contexts/form_validation_context';
-import { VisDataContext } from '../contexts/vis_data_context';
-
-const types = {
-  timeseries,
-  table,
-  metric,
-  top_n: topN,
-  gauge,
-  markdown,
-};
-
-const checkModelValidity = (validationResults) =>
-  Boolean(Object.values(validationResults).every((isValid) => isValid));
-
-export function PanelConfig(props) {
-  const { model } = props;
-  const Component = types[model.type];
-  const [formValidationResults] = useState({});
-  const [visData, setVisData] = useState({});
-
-  useEffect(() => {
-    model.isModelInvalid = !checkModelValidity(formValidationResults);
```

Comment:
it is not allowed to modify props, so this was converted to the [callback](https://github.com/elastic/kibana/pull/93231/files#diff-12c84fc25ef8ee816eab6f649d77e9b85fdc14d57b666fc47cde09393b04ea31R50)